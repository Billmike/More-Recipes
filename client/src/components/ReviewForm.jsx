import React from 'react';
import classNames from 'classnames';
import Loader from './Loader';

const ReviewForm = (props) => {
  if (props.isLoading) {
    return <Loader />;
  }
  return (
    <form
      className="form-group container form-container"
      onSubmit={props.reviewRecipe}
      id="form"
    >
      <div>
        <textarea
          id="reviewFormId"
          className={classNames('form-control review-form', {
            'has-errors': props.errors.name
          })}
          value={props.reviewText}
          onChange={props.onChange}
          placeholder="Enter a review for this recipe"
        />
        {props.errors.name && (
          <span className="help-block has-errors">{props.errors.name}</span>
        )}
      </div>
      <button
        id="submit-review"
        className="btn btn-primary btn-lg btn-block form-button"
      >
        Submit review
      </button>
    </form>
  );
};

export default ReviewForm;
