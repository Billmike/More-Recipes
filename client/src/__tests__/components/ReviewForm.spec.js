import React from 'react';
import { shallow } from 'enzyme';
import ReviewForm from '../../components/ReviewForm';

describe('<ReviewForm />', () => {
  it('Should render the review form correctly', () => {
    const wrapper = shallow(<ReviewForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
