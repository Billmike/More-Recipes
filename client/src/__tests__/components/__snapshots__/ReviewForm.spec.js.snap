// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewForm /> Should render the review form correctly 1`] = `
<form
  className="form-group container form-container"
  id="form"
>
  <textarea
    className="form-control review-form"
    id="reviewFormId"
    placeholder="Enter a review for this recipe"
  />
  <button
    className="btn btn-primary btn-lg btn-block form-button"
    id="submit-review"
  >
    Submit review
  </button>
</form>
`;
